test/e2e/platform/platform_test.go

```diff
@@ -37,65 +39,248 @@ import (
+			})
+		})
+
+		It("Delete Baremetal cluster", func() {
+			Expect(deleteCluster(clsName)).Should(Succeed())
+
+			out, _ := runCmd("kubectl get clusters --kubeconfig " + tkeKubeConfigFile)
+			Expect(out).ShouldNot(ContainSubstring(clsName))
+		})
 	})
 
-	It("complete cluster lifecycle", func() {
-		By("create baremetal cluster")
-		restconf, err := t.GetKubeConfig()
-		Expect(err).To(BeNil())
-		client := tkeclientset.NewForConfigOrDie(restconf)
-		err = createBaremetalCluster(context.Background(), client, masterNodes, workerNodes)
-		Expect(err).To(BeNil())
+	Context("Imported cluster", func() {
+		var cluster *platformv1.Cluster
+		var credential platformv1.ClusterCredential
+
+		BeforeEach(func() {
+			if cluster == nil {
+				// Prepare a cluster to be imported
+				cluster, err = createCluster(1)
+				Expect(err).To(BeNil())
+
+				// Get the credential of cluster
+				credentials, err := tkeClient.PlatformV1().ClusterCredentials().List(context.Background(), metav1.ListOptions{
+					FieldSelector: "clusterName=" + cluster.Name,
+				})
+				Expect(credentials.Items).Should(HaveLen(1))
+				Expect(err).Should(BeNil())
+				credential = credentials.Items[0]
+
+				// Delete cluster from the global cluster to be imported
+				out, _ := runCmd("kubectl delete cluster " + cluster.Name + " --kubeconfig " + tkeKubeConfigFile)
+				Expect(out).Should(ContainSubstring("deleted"))
+			}
+		})
+
+		It("Import cluster", func() {
+			// Import cluster
+			importedCluster, err := importCluster(cluster.Spec.Machines[0].IP, 6443, credential.CACert, credential.Token)
+			Expect(err).Should(BeNil())
+			Expect(importedCluster.Name).ShouldNot(Equal(cluster.Name))
+			Expect(importedCluster.Status.Phase).Should(Equal(platformv1.ClusterRunning))
+			Expect(importedCluster.Spec.Type).Should(Equal("Imported"))
+
+			out, _ := runCmd("kubectl get clusters --kubeconfig " + tkeKubeConfigFile + " | grep " + importedCluster.Name)
+			Expect(out).Should(ContainSubstring("Running"))
+		})
 	})
 })
 
-func createBaremetalCluster(ctx context.Context, client *tkeclientset.Clientset, masterNodes []cloudprovider.Instance,
-	workerNodes []cloudprovider.Instance) error {
-	time.Sleep(70 * time.Second)
-	cluster := &platformv1.Cluster{
+func createCluster(masterNodeNum int64) (cluster *platformv1.Cluster, err error) {
+	masterNodes, err := provider.CreateInstances(masterNodeNum)
+	if err != nil {
+		return nil, err
+	}
+	time.Sleep(10 * time.Second)
+	return createClusterWithMasterNodes(masterNodes)
+}
+
+func createClusterWithMasterNodes(masterNodes []cloudprovider.Instance) (cluster *platformv1.Cluster, err error) {
+	klog.Info("Create cluster")
+	cluster = &platformv1.Cluster{
 		Spec: platformv1.ClusterSpec{
-			Type:     "Baremetal",
-			Features: platformv1.ClusterFeature{EnableMasterSchedule: true},
+			Type:          "Baremetal",
+			Features:      platformv1.ClusterFeature{EnableMasterSchedule: true},
+			Version:       "1.18.3",
```

Comment:
Hard code cluster version is not a good way. If next release doesn't support this version, test will fail.

Reply:
nice catch!  cool!

Reply:
pkg/spec/spec.go has K8sVersions to fix it, please use the latest version for e2e test

Reply:
Got it. Thx